### ISSUE TITLE
Reducer Composition

### ISSUE THREAD

**Issue by dawidvdh** [erikras/ducks-modular-redux#41]
Hi, I have been using the ducks structure for quite a while now. But i find myself in a position where i need to compose reducers (categories -> posts), what is the recommended structure for composition?

**Comment by Rewieer** [comment]
I'm not quite sure about what you want to do. Can you rephrase ?

**Comment by dawidvdh** [comment]
sure thing, so [https://github.com/reactjs/redux/blob/master/examples/todos-flow/src/reducers/todos.js#L28](Take the Todo Example), its composing multiple reducers so for example the todo reducer is also composing the createTodo and toggleTodo, how does one do this using the ducks structure?

**Comment by Rewieer** [comment]
Technically it's not composing the createTodo, he just uses the createTodo with the same signature of a reducer, which may look confusing.  Same goes for toggle todo.
In the following snippet you'll see i'm not using such a function to create nor update.
`import * as actions from './actions';
import lodash from 'lodash';

import { combineReducers } from 'redux';

const allIds = (state = [], action) => {
  switch(action.type){
    case actions.POST_FETCH:
      return [
        ...state,
        ...action.ids
      ];
    case actions.POST_CREATE:
      return [
        ...state,
        action.post.id
      ];
    case actions.POST_SET_SERVER_ID:
      return state.map(id => id !== action.previousId ? id : action.nextId);
    case actions.POST_DELETE:
      return state.filter(id => id !== action.id);

    default: return state;
  }
};

const byId = (state = {}, action) => {
  switch(action.type){
    case actions.POST_FETCH:
      return {
        ...state,
        ...action.posts
      };
    case actions.POST_CREATE:
    case actions.POST_UPDATE:
      return {
        ...state,
        [action.post.id]: action.post
      };
    case actions.POST_SET_SERVER_ID:{
      if (!state.hasOwnProperty(action.previousId)) {
        return state;
      }

      const nextState = lodash.merge({}, state);
      const post = nextState[action.previousId];

      delete nextState[action.previousId];

      nextState[action.nextId] = {
        ...post,
        id: action.nextId
      };

      return nextState;
    }
    /*
     * Brackets are added because it seems that variables having the same name
     * even in difference cases causes bugs.
     *
     */
    case actions.POST_DELETE:{
      if (!state.hasOwnProperty(action.id)) {
        return state;
      }

      const nextState = lodash.merge({}, state);
      delete nextState[action.id];

      return nextState;
    }
    default: return state;
  }
};

export default combineReducers({
  allIds,
  byId
});`

**Comment by SeanRoberts** [comment]
Typically I will just write the two separate reducers and then `export default combineReducers({ createTodo, toggleTodo })` from the duck file.
